projects/core/src/datetimepicker/calendar.ts

```diff
@@ -525,6 +599,50 @@ export class MatDatetimepickerCalendar<D> implements AfterContentInit, OnDestroy
     event.preventDefault();
   }
 
+  /** Handles keydown events on the calendar body when calendar is in multi-year view. */
+  private _handleCalendarBodyKeydownInMultiYearView(event: KeyboardEvent): void {
+    // tslint:disable-next-line:deprecation
+    switch (event.keyCode) {
```

Comment:
hm.. this is rather ugly. but, maybe acceptable.

Reply:
Hey bro, I followed the code guideline like line 508 _handleCalendarBodyKeydownInMonthView if the code is ugly means that the guideline is ugly, :-)

Reply:
Hey bro, I followed the code guideline like line 508 _handleCalendarBodyKeydownInMonthView if the code is ugly means that the guideline is ugly, :-)

Reply:
well it's not exactly mine :D